From 98a22314fcf9a4b9ef57bbd2b44a94c2a7781e16 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 6 May 2024 14:35:17 +0200
Subject: Do not use std::format in ReadAnythingAppModel

std::format is still not allowed in Chrome (and this usage also broke
libstdc++ build as it was missing the required include). Fix by
using base::StringPrintf.

Bug: 41455655
Change-Id: I9a99e350c6ebf5e99baf5f90c06be6bad9c2a106
---
 chrome/renderer/accessibility/read_anything_app_model.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/chrome/renderer/accessibility/read_anything_app_model.cc b/chrome/renderer/accessibility/read_anything_app_model.cc
index 25aa4c88ba222..ab459a44de290 100644
--- a/chrome/renderer/accessibility/read_anything_app_model.cc
+++ b/chrome/renderer/accessibility/read_anything_app_model.cc
@@ -11,6 +11,7 @@
 #include "base/check.h"
 #include "base/containers/contains.h"
 #include "base/metrics/histogram_functions.h"
+#include "base/strings/stringprintf.h"
 #include "content/public/renderer/render_thread.h"
 #include "services/strings/grit/services_strings.h"
 #include "ui/accessibility/accessibility_features.h"
@@ -1059,7 +1060,7 @@ std::string ReadAnythingAppModel::GetHtmlTag(
     int32_t hierarchical_level =
         ax_node->GetIntAttribute(ax::mojom::IntAttribute::kHierarchicalLevel);
     if (hierarchical_level) {
-      return std::format("h{}", hierarchical_level);
+      return base::StringPrintf("h%" PRId32, hierarchical_level);
     }
   }
 
@@ -1159,7 +1160,7 @@ std::string ReadAnythingAppModel::GetHeadingHtmlTagForPDF(
   int32_t hierarchical_level =
       ax_node->GetIntAttribute(ax::mojom::IntAttribute::kHierarchicalLevel);
   if (hierarchical_level) {
-    return std::format("h{}", hierarchical_level);
+    return base::StringPrintf("h%" PRId32, hierarchical_level);
   }
   return html_tag;
 }
-- 
2.40.1

